src/components/timeline/TrackBandwidthGraph.js

```diff
@@ -0,0 +1,690 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import { InView } from 'react-intersection-observer';
+import { Localized } from '@fluent/react';
+import { withSize } from 'firefox-profiler/components/shared/WithSize';
+import {
+  getStrokeColor,
+  getFillColor,
+} from 'firefox-profiler/profile-logic/graph-color';
+import explicitConnect from 'firefox-profiler/utils/connect';
+import {
+  formatBytes,
+  formatNumber,
+} from 'firefox-profiler/utils/format-numbers';
+import { bisectionRight } from 'firefox-profiler/utils/bisect';
+import {
+  getCommittedRange,
+  getCounterSelectors,
+  getPreviewSelection,
+  getProfileInterval,
+} from 'firefox-profiler/selectors/profile';
+import { getThreadSelectors } from 'firefox-profiler/selectors/per-thread';
+import { Tooltip } from 'firefox-profiler/components/tooltip/Tooltip';
+import {
+  TooltipDetails,
+  TooltipDetail,
+  TooltipDetailSeparator,
+} from 'firefox-profiler/components/tooltip/TooltipDetails';
+import { EmptyThreadIndicator } from './EmptyThreadIndicator';
+import { TRACK_BANDWIDTH_COLOR } from 'firefox-profiler/app-logic/constants';
+import { getSampleIndexRangeForSelection } from 'firefox-profiler/profile-logic/profile-data';
+import { co2 } from '@tgwf/co2';
+
+import type {
+  CounterIndex,
+  Counter,
+  Thread,
+  ThreadIndex,
+  AccumulatedCounterSamples,
+  Milliseconds,
+  PreviewSelection,
+  CssPixels,
+  StartEndRange,
+  IndexIntoSamplesTable,
+} from 'firefox-profiler/types';
+
+import type { SizeProps } from 'firefox-profiler/components/shared/WithSize';
+import type { ConnectedProps } from 'firefox-profiler/utils/connect';
+
+import './TrackBandwidth.css';
+
+/**
+ * When adding properties to these props, please consider the comment above the component.
+ */
+type CanvasProps = {|
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +counter: Counter,
+  +counterSampleRange: [IndexIntoSamplesTable, IndexIntoSamplesTable],
+  +accumulatedSamples: AccumulatedCounterSamples,
+  +maxCounterSampleCountPerMs: number,
+  +interval: Milliseconds,
+  +width: CssPixels,
+  +height: CssPixels,
+  +lineWidth: CssPixels,
+|};
+
+/**
+ * This component controls the rendering of the canvas. Every render call through
+ * React triggers a new canvas render. Because of this, it's important to only pass
+ * in the props that are needed for the canvas draw call.
+ */
+class TrackBandwidthCanvas extends React.PureComponent<CanvasProps> {
```

Comment:
I thought a bit about how we could factor out some code to reduce code duplication. I think one thing we could relatively easily do without needing to go all the way and support generic counters would be to extract a TrackCountercanvas object from this. The parent component would need to provide the getY function which is specific for some counters, and that function would be reused by the function drawing the dot.
The power track, memory track and bandwidth track could share this track canvas implementation. I think we could do this as a follow-up, and the motivation for the follow-up could be to speed up drawing the memory chart, which still doesn't use the max-min decimation algorithm that was put in place for power tracks (and copied here).

Re-using the same canvas component for custom marker tracks might be slightly harder as these support markers with a duration.